packages/lib-sourcify/src/Verification/Verification.ts

```
@@ -0,0 +1,500 @@
+        lt(this.compilation.compilerVersion, '0.8.21') &&
+        !settings.optimizer?.enabled &&
+        settings.viaIR
+      ) {
+        logInfo('Force Emscripten compiler', {
+          address: this.address,
+          chainId: this.sourcifyChain.chainId,
+        });
+
+        // Try to verify again with Emscripten
+        return SolidityBugType.ECMASCRIPT_BUG;
+      }
+
+      // Case when extra unused files in compiler input cause different bytecode
+      // See issues:
+      //   https://github.com/ethereum/sourcify/issues/618
+      //   https://github.com/ethereum/solidity/issues/14250
+      //   https://github.com/ethereum/solidity/issues/14494
+      try {
+        const [, deployedAuxdata] = splitAuxdata(
+          this.onchainRuntimeBytecode,
+          AuxdataStyle.SOLIDITY,
+        );
+        const [, recompiledAuxdata] = splitAuxdata(
+          compiledRuntimeBytecode,
+          AuxdataStyle.SOLIDITY,
+        );
+        // Metadata hashes match but bytecodes don't match.
+        if (
+          deployedAuxdata === recompiledAuxdata &&
+          settings.optimizer?.enabled
+        ) {
+          throw new VerificationError(
+            "It seems your contract's metadata hashes match but not the bytecodes. You should add all the files input to the compiler during compilation and remove all others. See the issue for more information: https://github.com/ethereum/sourcify/issues/618",
+            'EXTRA_FILE_INPUT_BUG',
+          );
+        }
+      } catch (e: any) {
+        if (e instanceof VerificationError) {
+          throw e;
+        }
+        logWarn('Error checking for extra-file-input-bug', {
+          chain: this.sourcifyChain.chainId,
+          address: this.address,
+          error: e.message,
+        });
+        throw new VerificationError(
+          e.message,
+          'FAILED_TO_CHECK_EXTRA_FILE_INPUT_BUG',
+        );
+      }
```

Why do we have this explicit check? Is there a known case of such errors? Seemed unnecessary to me

---

Yes I think that's a little unnecessary. I don't think it's likely anything will go wrong here. We can propagate the error without catching or we could have a catch-all clause with a generic VerificationError. `FAILED_TO_CHECK_EXTRA_FILE_INPUT_BUG` seemed a bit too much

---

I'll just remove the error for now